sockeye/optimizers.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not
+# use this file except in compliance with the License. A copy of the License
+# is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""
+Extra optimizers not included in MXNet.
+"""
+
+from abc import abstractmethod
+from collections import namedtuple
+import math
+
+from mxnet.ndarray import NDArray, sqrt, zeros
+from mxnet.optimizer import Optimizer
+
+from sockeye.utils import check_condition
+
+BatchState = namedtuple("BatchState", ["metric_val"])
+CheckpointState = namedtuple("CheckpointState", ["checkpoint", "metric_val"])
+
+
+class SockeyeOptimizer(Optimizer):
+    """
+    Optimizer that has access to additional information from the last batch and the last chekpoint
+    when updating weights.
+    """
+    def __init__(self, **kwargs):
```

Comment:
could you try adding optimizers.py to typechecked-files in the package root?
You will need to annotate the return type of __init__ methods with "-> None" to avoid an error.